Comp2023/src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -171,6 +206,69 @@ public double getDistLimelight( )
     return m_distLL;
   }
 
+  public double getTargetID( )
+  {
+    return m_targetID;
+  }
+
+  public double getTargetLatency( )
+  {
+    return m_targetLatency;
+  }
+
+  public Transform3d getBotPoseTransform3d( )
+  {
+    return m_botposeTransform3d;
+  }
+
+  public Pose2d getBotPose2d(boolean checkPosition)
+  {
+    Translation2d tran;
+    Rotation2d rot;
+
+    // Perform valid checks and return null if it doesn't exist
+    if (m_targetValid)
+    {
+      tran = new Translation2d(m_botposeTransform3d.getX( ), m_botposeTransform3d.getY( ));
+      double degrees = m_yawBotPose + ((m_yawBotPose < 0) ? 360 : 0);
+      rot = new Rotation2d(Units.degreesToRadians(degrees));
+      if (checkPosition)
+        if (isBotPoseValid(tran))
+        {
+          return new Pose2d(tran, rot);
+        }
+      if (!checkPosition)
+      {
+        return new Pose2d(tran, rot);
+      }
+
+    }
+
+    // DataLogManager.log(getSubsystem( )                                         //  
+    //     + "VISION: Radians sent " + rot                                        //
+    //     + " from the limelight " + m_botposeTransform3d.getRotation( ).getZ( ) //
+    //     + " Angle : " + m_botposeTransform3d.getRotation( ).getAngle( )        //
+    // );
+
+    return null;
+  }
+
+  public boolean isBotPoseValid(Translation2d botpose)
+  {
+    //Only adding vision measurements that are already within one meter or so of the current pose estimate
+    Translation2d currentPose = RobotContainer.getInstance( ).m_swerve.m_poseEstimator.getEstimatedPosition( ).getTranslation( );
+
+    if (Math.abs(currentPose.getX( ) - botpose.getX( )) <= 1)

```

Comment:
Would be more obvious if we did a pose subtraction (minus) and then do the absolute value of the X and Y component.